aesara/link/numba/dispatch.py

```diff
@@ -0,0 +1,76 @@
+from functools import singledispatch
+
+import numba
+
+from aesara.graph.fg import FunctionGraph
+from aesara.link.utils import fgraph_to_python
+from aesara.scalar.basic import Composite, ScalarOp
+from aesara.tensor.elemwise import Elemwise
+
+
+@singledispatch
+def numba_typify(data, dtype=None, **kwargs):
+    return data
+
+
+@singledispatch
+def numba_funcify(op, **kwargs):
+    """Create a Numba compatible function from an Aesara `Op`."""
+    raise NotImplementedError(f"No Numba conversion for the given `Op`: {op}")
+
+
+@numba_funcify.register(FunctionGraph)
+def numba_funcify_FunctionGraph(
+    fgraph,
+    order=None,
+    input_storage=None,
+    output_storage=None,
+    storage_map=None,
+    **kwargs,
+):
+    return fgraph_to_python(
+        fgraph,
+        numba_funcify,
+        numba_typify,
+        order,
+        input_storage,
+        output_storage,
+        storage_map,
+        fgraph_name="numba_funcified_fgraph",
+        **kwargs,
+    )
+
+
+@numba_funcify.register(ScalarOp)
+def numba_funcify_ScalarAdd(op, **kwargs):
+    import numpy as np
+
+    numpy_func = getattr(np, op.nfunc_spec[0])
+
+    @numba.njit
+    def scalar_func(*args):
+        result = args[0]
+        for arg in args[1:]:
+            result = numpy_func(arg, result)
```

Comment:
Is this an instance in which the Numba conversion was having trouble dealing with varargs (e.g. `numpy_func(*args)`)?

Reply:
That error occurs when we put a `numba.vectorize` tag on a function which (or it's child functions) may have varargs like `*args` . It's the `TODO` listed in `Elemwise` 's Numba conversion.